### PR TITLE
Finish component

### PR DESCRIPTION
**Pull request by sammygriffiths**
Added a 'finished component' as per https://github.com/Scoombe/react-typing/issues/3 that looks like the this:

image

This will also require https://github.com/Scoombe/wpmTest/pull/28

### CODE REVIEW

**Comment by Scoombe** [comment]
@sammygriffiths it doesn't work properly after the reset, I don't think the timers have been reset properly. Shall I take a look? @sammygriffiths

**Comment by Scoombe** [comment]
@sammygriffiths the finish function isn't called  either.

**Comment by sammygriffiths** [comment]
The timer won't reset properly until the other PR I made is merged (https://github.com/Scoombe/wpmTest/pull/28). That is to reset the timer in the `restartTest` function of the `wordsPerMinTest` class.

And it looks to me like the finish function is called, isn't it? It's passed to the `wordsPerMinTest` class on line 13 (I haven't actually changed that bit).

**Comment by sammygriffiths** [comment]
It might be best if you could have a look at https://github.com/Scoombe/wpmTest/pull/28 first as I believe that when that PR is merged it will fix the issues you mention here.

I did initially do all the work in this repo but I thought it made sense to reset the timer in the other repo as the `restartTest` sounds like it should reset the timer too (in my opinion).

File: src/Components/TypingTest.js [hunk 1]
```diff
@@ -101,9 +107,31 @@ class TypingTest extends Component {
...
+        this.setDisplayText(false);
+      } else {
+        this.setDisplayText(false, charCheck.errorText);
+      }
+    }
+  }
+
   finishedFunction() {
     this.wordsTest.started = false;
-    this.setState({ displayString: 'You have finished' });
```
Comment (Scoombe) [inline src/Components/TypingTest.js]: ad the line 

`
this.wordsTest.finishStopWatch();
`

on the finished function

Comment (Scoombe) [reply src/Components/TypingTest.js]: I have merged your pull request to wpm test. I was looking at your comments out of order and I didn't really look at your pull request for wpm test. My bad.

Comment (sammygriffiths) [reply src/Components/TypingTest.js]: No problem @Scoombe , I've actually made another PR on wpm test as the TS needed building, that then should allow this PR to work :)

Comment (sammygriffiths) [reply src/Components/TypingTest.js]: https://github.com/Scoombe/wpmTest/pull/30

**Comment by Scoombe** [comment]
@sammygriffiths  could you have a look at separating the components one test component and another separate finish component that is displayed on the TypingTest Page thanks.

**Comment by sammygriffiths** [comment]
@Scoombe Yea that's a good idea.
Do you think maybe this should be merged and that should be a part of https://github.com/Scoombe/react-typing/issues/4 though as it probably counts as refactoring?

That's my opinion but happy with whatever you think is best 😄

**Comment by Scoombe** [comment]
@sammygriffiths Ok, just fix the linting then

**Comment by Scoombe** [comment]
@sammygriffiths Merged. Thanks for your help!

**Comment by sammygriffiths** [comment]
Apologies I only just got around to checking this and saw your comment about linting, but it looks like you beat me to it 😛. 
No problem, I have more ideas for it too if you're open to more PRs when I get chance.

**Pull request merged**